core/server/master/src/main/java/alluxio/master/meta/MetaMasterSync.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.Configuration;
+import alluxio.PropertyKey;
+import alluxio.heartbeat.HeartbeatExecutor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Task that carries out the necessary standby master to leader master communications, including
+ * register and heartbeat. This class manages its own {@link MetaMasterMasterClient}.
+ *
+ * When running, this task first requests master configuration from the standby master,
```

Comment:
this sounds weird since the task is a part of the standby master, so the master would be requesting configuration from itself. Can we re-phrase?

Reply:
Re-phrased, PTAL at the new one. Thanks